### DISCUSSION TITLE
[question] How to correctly split into layers

### DISCUSSION THREAD

**Discussion by II11II** [serverpod/serverpod#378] [Q&A] [answered]
Following the clean architecture, I would like to split it into data, domain, and presentation layers.
Is it considered in Serverpod architecture to apply architecture patterns from the user side?

**Comment by vlidholt** [comment, marked as answer]
Serverpod by design naturally divides your architecture into a presentation / domain / data layering. To make the architecture even more clean, it's recommended that you use a business layer between your endpoint methods and the database. This allows you to use the same code for similar tasks in different endpoint methods.
